File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int lo = 0;
        int hi = height.length-1;
        int res = 0;
        while(lo < hi){
            res = Math.max(res, (hi-lo)*Math.min(height[lo], height[hi]));
            if(height[lo] <= height[hi]){
                lo += 1;
            }
            else{
                hi -= 1;
            }
        }
        return res;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        res = 0
        n = len(height)
        lt = 0
        rt = n-1
        while lt < rt:
            res = max(res, min(height[lt], height[rt])*(rt-lt))
            if height[lt] < height[rt]:
                lt += 1
            else:
                rt -= 1
        return res
            
        
        